src/utils/pagination/pagination.go

```diff
@@ -0,0 +1,90 @@
+package pagination
+
+import (
+	"fmt"
+	"github.com/dana-team/platform-backend/src/middleware"
+	"github.com/dana-team/platform-backend/src/types"
+	"github.com/dana-team/platform-backend/src/utils"
+	"github.com/gin-gonic/gin"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strconv"
+)
+
+const (
+	envDefaultPaginationLimit = "DEFAULT_PAGINATION_LIMIT"
+	defaultPaginationLimit    = 100
+)
+
+// BuildListOptions builds the ListOptions based on page and limit
+func buildListOptions(limit int64, continueToken string) v1.ListOptions {
+	return v1.ListOptions{
+		Limit:         limit,
+		Continue:      continueToken,
+		LabelSelector: utils.ManagedLabelSelector,
+	}
+}
+
+// FetchPage fetches the specified page with given limit
+func FetchPage[T any](limit, page int, paginator Paginator[types.List[T]]) ([]T, error) {
```

Comment:
Ill change the name of the function to fetch resources from page this empties you fetching page number not k8s resources content in specific page